index.js

```diff
@@ -18,10 +22,22 @@ class Chapter {
 
   map(array, callback) {
     const output = [];
-    for (let i = 0; i < array.length; i++) {
-      output.push(callback(array[i]));
+
+    if (typeof array === 'function') {
+      for (let i = 0; i < this.result.length; i++) {
```

Comment:
we can simplify this loop a bit using `for .. of`

Reply:
Done